cmd/ketch/pool_remove.go

```diff
@@ -35,14 +36,28 @@ type poolRemoveOptions struct {
 }
 
 func poolRemove(ctx context.Context, cfg config, options poolRemoveOptions, out io.Writer) error {
-	pool := ketchv1.Pool{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: options.Name,
-		},
+	var pool ketchv1.Pool
+	var ns corev1.Namespace
+
+	if err := cfg.Client().Get(ctx, types.NamespacedName{Name: options.Name}, &pool); err != nil {
+		return fmt.Errorf("failed to get pool: %w", err)
 	}
+
+	// Get namespace that was created with the pool so it can be removed
+	if err := cfg.Client().Get(ctx, types.NamespacedName{Name: pool.Spec.NamespaceName}, &ns); err != nil {
+		return fmt.Errorf("failed to get namespace: %w", err)
+	}
+
 	if err := cfg.Client().Delete(ctx, &pool); err != nil {
```

Comment:
We already pass pool in `cfg.Client().Delete` func. Is it possible to put namespace removal logic inside it instead of here?

Reply:
`cfg.Client().Delete` is a function from a dependency, the Kubebuilder controller-runtime library 🙂 

https://github.com/kubernetes-sigs/controller-runtime/blob/cf7ac887d082b481ec1389ddcd576f4140b15595/pkg/client/client.go#L167

Unless I misunderstood?